app/src/main/java/com/nativo/sampleapp/ViewAdapter/ListViewAdapter.kt

```diff
@@ -0,0 +1,159 @@
+import net.nativo.sdk.injectable.NtvInjectableType
+
+class ListViewAdapter(private val context: Context, private val listView: ViewGroup) : BaseAdapter(), NtvSectionAdapter {
+
+    companion object {
+        const val ITEM_COUNT = 30
+    }
+
+    var integerList: MutableList<Int> = ArrayList()
+    var initialNativoRequest = true
+
+    init {
+        // Nativo init
+        NativoSDK.initSectionWithAdapter(this, SECTION_URL, context)
+
+        for (i in 0..ITEM_COUNT) {
+            integerList.add(i)
+        }
+    }
+
+    override fun getCount(): Int {
+        return integerList.size
+    }
+
+    override fun getItem(i: Int): Any? {
+        return null
+    }
+
+    override fun getItemId(i: Int): Long {
+        return 0
+    }
+
+    override fun getView(i: Int, view: View?, viewGroup: ViewGroup): View {
+        val convertView: View = view
+            ?: LayoutInflater.from(viewGroup.context)
+                .inflate(R.layout.publisher_article, viewGroup, false)
+        bindView(convertView, integerList[i], i)
+        return convertView
+    }
+
+    private fun bindView(container: View, item: Int, i: Int) {
+        val articleImage = container.findViewById<ImageView>(R.id.article_image)
+        val sponsoredIndicator = container.findViewById<ImageView>(R.id.sponsored_ad_indicator)
+        val articleAuthor = container.findViewById<TextView>(R.id.article_author)
+        val articleTitle = container.findViewById<TextView>(R.id.article_title)
+        val sponsoredTag = container.findViewById<TextView>(R.id.sponsored_tag)
+
+        articleImage?.setImageResource(R.drawable.newsimage)
+        sponsoredIndicator?.visibility = View.INVISIBLE
+        articleAuthor?.setText(R.string.sample_author)
+        articleTitle?.setText(R.string.sample_title)
+        sponsoredTag?.visibility = View.INVISIBLE
+
+        container.setOnClickListener {
+            context.startActivity(
+                Intent(
+                    Intent.ACTION_VIEW,
+                    Uri.parse(CLICK_OUT_URL)
+                )
+            )
+        }
+
+        if (shouldShowPlaceAd(item)) {
+            val isAdContentAvailable =
+                NativoSDK.placeAdInView(container, listView, SECTION_URL, i, null)
+            Log.w(NtvTAG, "isAdContentAvailable = $isAdContentAvailable")
+        }
+    }
+
+    private fun shouldShowPlaceAd(item: Int): Boolean {
+        return item % 3 == 1
+    }
+
+    override fun didReceiveAd(didGetFill: Boolean, inSection: String) {
+        Log.d(NtvTAG, "Did receive ad: $didGetFill")
+
+        if (didGetFill && initialNativoRequest) {
+            Log.w(NtvTAG, "Needs Reload Everything")
+            notifyDataSetChanged()
+        }
+
+        initialNativoRequest = false
+    }
+
+    override fun didAssignAdToLocation(
+        location: Int,
+        adData: NtvAdData,
+        inSection: String,
+        container: ViewGroup
+    ) {
+        Log.d(NtvTAG, "didAssignAdToLocation: $location")
```

Comment:
When an ad gets assigned to a location, you should add an extra row to your datasource(integerList) and then inject that row here